massa-pool/src/tests/scenario.rs

```diff
@@ -183,6 +184,172 @@ async fn test_pool() {
+            for lst in thread_tx_lists.iter_mut() {
+                lst.reverse();
+                lst.truncate(*max_pool_size_per_thread as usize);
+            }
+
+            // checks ops for thread 0 and 1 and various periods
+            for thread in 0u8..=1 {
+                for period in 0u64..70 {
+                    let target_slot = Slot::new(period, thread);
+                    let max_count = 3;
+                    let res = pool_command_sender
+                        .get_operation_batch(
+                            target_slot,
+                            OperationHashSet::default(),
+                            max_count,
+                            10000,
+                        )
+                        .await
+                        .unwrap();
+                    assert!(res
+                        .iter()
+                        .map(|(id, op, _)| (id, op.to_bytes_compact().unwrap()))
+                        .eq(thread_tx_lists[target_slot.thread as usize]
+                            .iter()
+                            .filter(|(_, _, r)| r.contains(&target_slot.period))
+                            .take(max_count)
+                            .map(|(id, op, _)| (id, op.to_bytes_compact().unwrap()))));
+                }
+            }
+            // op ending before or at period 45 should be discarded
+            let final_period = 45u64;
+            pool_command_sender
+                .update_latest_final_periods(vec![final_period; *thread_count as usize])
+                .await
+                .unwrap();
+            for lst in thread_tx_lists.iter_mut() {
+                lst.retain(|(_, op, _)| op.content.expire_period > final_period);
+            }
+            // checks ops for thread 0 and 1 and various periods
+            for thread in 0u8..=1 {
+                for period in 0u64..70 {
+                    let target_slot = Slot::new(period, thread);
+                    let max_count = 4;
+                    let res = pool_command_sender
+                        .get_operation_batch(
+                            target_slot,
+                            OperationHashSet::default(),
+                            max_count,
+                            10000,
+                        )
+                        .await
+                        .unwrap();
+                    assert!(res
+                        .iter()
+                        .map(|(id, op, _)| (id, op.to_bytes_compact().unwrap()))
+                        .eq(thread_tx_lists[target_slot.thread as usize]
+                            .iter()
+                            .filter(|(_, _, r)| r.contains(&target_slot.period))
+                            .take(max_count)
+                            .map(|(id, op, _)| (id, op.to_bytes_compact().unwrap()))));
+                }
+            }
+            // add transactions from protocol with a high fee but too much in the future: should be ignored
```

Comment:
I think it would be clearer to write "Add transactions that should be ignored despite their high fees, due to them being too far in the future".